MetaMask/metamask-extension
# feat: Add new privacy policy toast

## **Description**

Changes:
Adds a new privacy policy banner that will show up after June 4, 2024.
If the user closes it manually, the privacy policy banner will not show up again.
If the user does not close the toast manually, the privacy policy banner will cease to show up after a day of being shown for the first time.
Changes the Survey banner so that we can display multiple banners at the same time without overlapping each other.

Considerations:
- Make sure this doesn't show up during onboarding
- This toast will show up even after years of the privacy policy being changed. We should set up a deadline for that or eventually revert this PR.
- Check if copy is correct.

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/23838?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/2319

## **Manual testing steps**

1. Go to the home page
2. Check if there's a new privacy banner there (date should be edited in the code so that the privacy policy shows up earlier than May 6, 2024)
3. When closing it, it shouldn't show up again
4. If not closing it manually, the toast should not show up again after 1 day of being shown for the first time. This data is saved in localStorage so editing the timestamp there should be enough to test this feature out.

## **Screenshots/Recordings**



### **Before**



### **After**

<img width="1031" alt="Screenshot 2024-04-22 at 15 51 53" src="https://github.com/MetaMask/metamask-extension/assets/11148144/f900e27e-2000-44a2-bf9c-47ce8f601821">
<img width="499" alt="Screenshot 2024-04-22 at 15 52 06" src="https://github.com/MetaMask/metamask-extension/assets/11148144/19410fc4-d0e1-4b60-a03f-e009d91df006">
<img width="354" alt="Screenshot 2024-04-22 at 15 52 20" src="https://github.com/MetaMask/metamask-extension/assets/11148144/69c65ba2-3bf8-4d72-8aef-f891119292dd">

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
